### DISCUSSION TITLE
Add Availability to UI templates

### DISCUSSION THREAD

**Discussion by epenet** [home-assistant/architecture#1149] [General]
When templates were added to the UI, `Availability` was apparently purposefully left out:

> The core team decided it was OK to bring template helpers to the UI, but in a limited way. Some of the goals taken into account were: Prevent the UI from becoming overwhelming for users with all kinds of experience levels, and preventing from ending up with a lot of options in a configuration flow that hurts UX.
> Therefore, we added the bare minimum fields that will satisfy a guesstimate of 80-90% of the use cases. With YAML support still being available, all other use cases would still be possible.

I propose that `Availability` is added to the UI for at least some UI templates

Sample PRs:
- sensor in https://github.com/home-assistant/core/pull/126806
- binary_sensor in https://github.com/home-assistant/core/pull/126807

Related to https://community.home-assistant.io/t/add-availability-template-to-the-new-template-sensor-helper/621069
See also https://github.com/home-assistant/architecture/discussions/1148

**Comment by frenck** [comment]
Seems like we crossed paths there.

But as per <https://github.com/home-assistant/core/pull/126806#issuecomment-2379072519>, this proposal has been decided against.

../Frenck

**Reply by epenet** [reply]
Adding the full clarification here for reference:

> Small clarification.
>
> Bringing templates to the UI was quite a long discussion. The core team decided it was OK to bring template helpers to the UI, but in a limited way. Some of the goals taken into account were: Prevent the UI from becoming overwhelming for users with all kinds of experience levels, and preventing from ending up with a lot of options in a configuration flow that hurts UX.
>
> Therefore, we added the bare minimum fields that will satisfy a guesstimate of 80-90% of the use cases. With YAML support still being available, all other use cases would still be possible.
>
> Adding this, breaks with the decisions made at the time. I'm adding this PR to to core team meeting agenda to discuss.
>
> ../Frenck
